client/helpers.go

```diff
@@ -36,6 +36,27 @@ func ContextMultiplex(meta schema.ClientMeta) []schema.ClientMeta {
 	return clients
 }
 
+// ApiFilterContextMultiplex returns a list of clients for each context from the cq config
+func ApiFilterContextMultiplex(path string) func(meta schema.ClientMeta) []schema.ClientMeta {
+	return func(meta schema.ClientMeta) []schema.ClientMeta {
+		client := meta.(*Client)
+
+		// in kubernetes version below 1.4 paths is nil
+		if client.paths != nil {
+			if _, ok := client.paths[path]; !ok {
+				client.Logger().Warn("The resource is not supported by current version of k8s", "path", path)
+				return []schema.ClientMeta{}
+			}
+		}
+
+		clients := make([]schema.ClientMeta, 0)
+		for _, ctxName := range client.contexts {
+			clients = append(clients, client.WithContext(ctxName))
+		}
+		return clients
```

Comment:
```suggestion
		clients := make([]schema.ClientMeta, 0, len(client.contexts))
		for _, ctxName := range client.contexts {
			clients = append(clients, client.WithContext(ctxName))
		}
		return clients
```
[non-blocking]  It seems like we can do the slice memory allocation in one go here, since we know the number of items that will be inserted.